### ISSUE TITLE
Provide feature requests for Sphinx

### ISSUE THREAD

**Issue by piogor** [tango-controls/tango-doc#31]
The features identified are:

1. additional roles: 
- :audition: For listing expected aution of a deocumen like beginner, admin, developer, user
- :language: For listing programming languages the document apply to lie c++, python, java or any

2. possibility to generate pages with indexes for different roles for example directive:
```
.. index::
     :roles: term
```
Which will generate/display an index of terms only.

3. possibility to display a named toctree in some other document with some filter:
Let's assume we have a toctree in index.rst:
```
.. toctree::
   :name: mastertoc

   doc1
   doc2
```
And somewere in a doc3.rst:
```
.. insert_toc:: mastertoc
   :filter-role: audition
   :filter-value: admin
```
Then output from doc3.rst will render a table of contents with items from mastertoc which points to sections containing :audition:`admin`

**Comment by piogor** [comment]
Link to how-to defining custom roles in Sphinx:
https://doughellmann.com/blog/2010/05/09/defining-custom-roles-in-sphinx/

**Comment by piogor** [comment]
The thread from Sphinx dev group is here: https://groups.google.com/forum/#!topic/sphinx-dev/RZEYZfk1KQM

**Comment by piogor** [comment]
It is solved by #207 as direct filtering withing ToC...
